src/js/data-model.jsx

```diff
@@ -1,82 +1,153 @@
-var React = require('react'),
-    _ = require('underscore');
+var React = require('react');
+var _ = require('underscore');
 
 
-var router = require('./router'),
-    facts = require('./facts');
+var CollapsibleText = require('./collapsible-text');
+var Dqa = require('./dqa');
+var Table = require('./table');
+var resources = require('./resources');
 
 
-var DataModelObject = require('./data-model-object'),
-    FactLog = require('./fact-log'),
-    Comparators = require('./comparators');
-
-
-// Takes a set of facts representing whose entities make up a data model.
 var DataModel = React.createClass({
     propTypes: {
-        facts: React.PropTypes.array.isRequired,
+        name: React.PropTypes.string.isRequired,
+        version: React.PropTypes.string.isRequired,
+        tables: React.PropTypes.array.isRequired,
         label: React.PropTypes.string.isRequired,
-        doc: React.PropTypes.string,
-        resource: React.PropTypes.string,
+        description: React.PropTypes.string,
+        activeTable: React.PropTypes.string,
+        activeField: React.PropTypes.string,
+        url: React.PropTypes.string,
+        etlConventions: React.PropTypes.string,
+        siteComments: React.PropTypes.object,
+        dqa: React.PropTypes.object,
     },
 
-    render: function() {
-        var entities = facts.aggregate(this.props.facts);
-
-        var resource = this.props.resource;
-
-        var items = _.map(entities, function(item) {
-            // Check if the item is the selected resource.
-            var className = item.ident === resource ? 'active' : '',
-                url = router.reverse(item.domain, item.ident);
+    getDefaultProps: function() {
+        return {
+            description: '',
+            url: "#",
+        };
+    },
 
+    getInitialState: function() {
+        return {
+            collapsedStatus: {},
+            tableWidth:600,
+        };
+    },
+    
+    render: function() {
+        var self = this;
+
+        var description = this.props.description;
+        if (this.props.tables && this.props.tables.length>0 && self.props.activeTable) {
+            var table = _.find(this.props.tables, function(f) { return f.name === self.props.activeTable; });
```

Comment:
Format this on multiple lines for readability.

```javascript
var table = _.find(this.props.tables, function(f) {
    return f.name == self.props.activeTable;
});
```

Also, a more *correct* way of ensuring a closure uses the current scope is to call `function() {}.bind(this);`. I think it makes the block more obvious since rather than using an closure variable e.g. `self`. What are your thoughts? Again we should define a convention for this.